File: 0792-binary-search/0792-binary-search.cpp
class Solution {
public:
    int bs(vector<int>& b,int low,int high,int target){
        if(low > high)
        {
            return -1;
        }

        int mid = (low + high)/2;
        if(b[mid] == target)
        {
            return mid;
        }
        else if(target > b[mid])
        {
            return bs(b,mid+1,high,target);
        }
        else{
            return bs(b,low,mid-1,target);
        }
    }


    int search(vector<int>& nums, int target) {
        return bs(nums,0,nums.size() - 1,target); 
    }
};

File: 0792-binary-search/0792-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int l=0;
        int s=nums.length;
        int h=s-1;
        int mid;
        while(l<=h){
            mid = (l+h)/2;

            if(nums[mid] == target){
                return mid;
            }

            else if(target > nums[mid]){
                l = mid+1;
            }
            else{
                h=mid-1;
            }
        }

        return -1;
    }
}